### PR TITLE
Enable IW in web

### LINKED ISSUE #9717: Support interactive window in web extension
There's a non trivial amount of work to make the interactive window support web. Mostly around the commands using the local file system for import/export.

### PR DESCRIPTION
**Pull request by amunger**
Fixes #9717

One notable command that was not enabled is `execute selection` since it uses an execution service to get the applicable block of code from a selection and I haven't taken the time to convert that yet.

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

**Comment by rchiodo** [comment]
We should probably convert the interactive tests to also run for the web. I did that for ipywidget tests and remote tests. I think you could likely follow a similar pattern.

File: src/notebooks/controllers/notebookControllerManager.ts [hunk 1]
```diff
@@ -233,11 +233,7 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
                     ? this.notebook.notebookDocuments.find((item) => item.notebookType === notebookType)
                     : undefined;
             const controller = await this.createDefaultRemoteController(notebookType, notebook);
-            if (controller) {
-                return controller;
-            }
-            traceVerbose('No default remote controller, hence returning the active interpreter');
-            return this.createActiveInterpreterController(notebookType, resource);
```
Comment (DonJayamanne) [inline src/notebooks/controllers/notebookControllerManager.ts:240]: Isn't this a breaking change for desktop?
I.e. its possible that we'll never have a controller, when in the past we'd return the active interpreter.

**Review by DonJayamanne — CHANGES_REQUESTED** [review]
Looks like we've introduced a breaking change to not have an active interpreter controller for IW

File: src/notebooks/controllers/notebookControllerManager.ts (hunk 1 above)
Comment (amunger) [reply src/notebooks/controllers/notebookControllerManager.ts]: whoops, I misunderstood what `isLocalLaunch` meant, good catch

File: src/interactive-window/editor-integration/codeLensFactory.ts [hunk 2]
```diff
@@ -256,8 +263,13 @@ export class CodeLensFactory implements ICodeLensFactory {
                 Commands.DebugStop,
                 Commands.RunCellAndAllBelowPalette
             ];
-            fullCommandList = fullCommandList.filter((item) => !commandsToBeDisabledIfNotTrusted.includes(item));
+        } else if (IsWebExtension) {
```
[the remark this answers is not in the document]
Comment (amunger) [reply src/interactive-window/editor-integration/codeLensFactory.ts:266]: you mean a separate `if`? yeah that would probably make more sense

Comment (amunger) [inline src/interactive-window/editor-integration/codeLensFactory.ts:266]: need to inject the boolean IsWebExtension, this is just the symbol, which is always true

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**